packages/plexus/src/Digraph/SvgEdge.tsx

```diff
@@ -62,15 +64,29 @@ export default class SvgEdge<U = {}> extends React.PureComponent<TProps<U>> {
       },
       getProps(setOnEdge, layoutEdge, renderUtils)
     );
+
+    const [startX, startY] = pathPoints[0];
+    const [endX, endY] = pathPoints[pathPoints.length - 1];
+
+    const xOffset = (label?.length ?? 0) * 5;
+    const labelX = (startX + endX) / 2;
+    const labelY = (startY + endY) / 2;
+
     return (
-      <path
-        d={d}
-        fill="none"
-        vectorEffect="non-scaling-stroke"
-        markerEnd={markerEnd}
-        markerStart={markerStart}
-        {...customProps}
-      />
+      <g>
+        <path
+          d={d}
+          fill="none"
+          vectorEffect="non-scaling-stroke"
+          markerEnd={markerEnd}
+          markerStart={markerStart}
+          {...customProps}
+        />
+
+        <text x={labelX - xOffset} y={labelY} fill="#000" fontSize="1rem" fontWeight="bold">
```

Comment:
> fontSize="1rem" fontWeight="bold"

is this something that can be expressed via stylesheet? How do we style the content of the nodes? Maybe `label` should also be a DOM content, not a plain text?

If too much work, ok to keep as plain text in the first iteration, but I'd still like to know the answer to the styling question.

Reply:
The code here is for the edge/arrow, nodes can be customised using `renderNode` property which is already being used for this graph, if you're talking about similar customization for edges then the [documentation](https://github.com/jaegertracing/jaeger-ui/tree/main/packages/plexus#edges-layer) says it can be expensive

<img width="992" alt="Screenshot 2023-11-14 at 3 33 00 PM" src="https://github.com/jaegertracing/jaeger-ui/assets/28570857/1281d37a-74c1-4138-9da2-05b7a3b99a70">

We can still try a `renderEdge` method and see how it affects the performance